gocov/test.go

```diff
@@ -76,6 +77,9 @@ func runTests(args []string) error {
 
 	// Unique -coverprofile file names are used so that all the files can be
 	// later merged into a single file.
+	commandChan := make(chan *exec.Cmd, len(pkgs))
+	responseChan := make(chan error, len(pkgs))
+	doneChan := make(chan bool)
```

Comment:
You could drop `doneChan` and close `commandChan` immediately after sending all of the commands over it. Then the `runTest` function could be simplified to:

```go
func runTest(jobChan chan *exec.Cmd, resultChan chan error) {
    for cmd := range jobChan {
        resultChan <- cmd.Run()
    }
}
```

However, this leaves another problem: tests may continue running after another one has failed. I would prefer that the in-progress test processes complete before exiting. For that reason, I suggest using [golang.org/x/sync/errgroup](https://godoc.org/golang.org/x/sync/errgroup). It would look something like:

```go
commandChan := make(chan *exec.Cmd, len(pkgs))
// ... send commands to commandChan ...
g, ctx := errgroup.WithContext(context.Background())
for i := 0; i < runtime.NumCPU(); i++ {
    g.Go(func() error {
        for {
            select {
            case cmd, ok := <-commandChan:
                if !ok {
                    return nil
                }
                if err := cmd.Run(); err != nil {
                    return err
                }
            case <-ctx.Done():
                return ctx.Err()
            }
        }
    })
}
if err := g.Wait(); err != nil {
    return err
}
```